consensus/core/src/synchronizer.rs

```diff
@@ -970,6 +963,15 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
         Ok(())
     }
 
+    /// The number of rounds above the highest accepted round to allow fetching missing blocks
+    /// via the periodic synchronization. Any missing blocks of higher rounds are considered
+    /// too far in the future to fetch. This property is used only when it's
+    /// detected that the node has fallen behind on its commit compared to the rest of the network,
+    /// otherwise scheduler will attempt to fetch any missing block.
+    fn missing_block_round_threshold(&self) -> Round {
+        self.context.parameters.commit_sync_batch_size
+    }
```

Comment:
Don't have an issue with this refactoring, but eventually after GC is enabled in mainnet this won't be necessary anymore as the flow gets disabled due to https://github.com/MystenLabs/sui/pull/21610/files#diff-b4d4115c07b7df91ec737bd67bf5ddc33f0877f14ef84e0794a6089309918638R909

Reply:
Good point. I think my refactor is basically a no-op.